### DISCUSSION TITLE
What Query Log to check when you have 2 x AGH servers running

### DISCUSSION THREAD

**Discussion by iptvcld** [AdguardTeam/AdGuardHome#7209] [Q&A]
I currently have 2 DNS (AGH) running and each of my devices have both DNS servers configured (DNS 1, DNS 2).

When a site is being blocked and I want to allow it on network, I always have to guess which DNS server its on.  For example, I go to search the query log on DNS 1 and there no mention of that blocked site and then when i check DNS 2, i can see the client and site hit that DNS server. 

Issue is now, if i unblock that site from DNS 2 then once my DNS sync occurs, it will be over written as DNS 1 does not have that unblock (the change was made on DNS 2).

Should I be able to see all the blocks and activity from DNS server 1?

Thank you,

**Comment by Dynamic5912** [comment]
> I currently have 2 DNS (AGH) running and each of my devices have both DNS servers configured (DNS 1, DNS 2).
> 
> 
> 
> When a site is being blocked and I want to allow it on network, I always have to guess which DNS server its on.  For example, I go to search the query log on DNS 1 and there no mention of that blocked site and then when i check DNS 2, i can see the client and site hit that DNS server. 
> 
> 
> 
> Issue is now, if i unblock that site from DNS 2 then once my DNS sync occurs, it will be over written as DNS 1 does not have that unblock (the change was made on DNS 2).
> 
> 
> 
> Should I be able to see all the blocks and activity from DNS server 1?
> 
> 
> 
> Thank you,

Why are you running 2 AGH instance simultaneously?

**Reply by iptvcld** [reply]
For redundancy and load balance, it’s a common approach

**Reply by Dynamic5912** [reply]
Allow it on both instances then?

**Reply by iptvcld** [reply]
I noted the scenario and issue faced with that. 

When searching the query log, you need to search each log from each dns server (always a guess) and since I have sync enabled. DNS 1 is master. So if I make a change on dns 2, dns 1 will override dns 2. 

Even though let’s say the client/device is on dns 2, I now need to make the change on DNS 1 so the sync can occur over. Not a good solution and having more than 1 dns server is very common in all types or organizations

**Reply by Dynamic5912** [reply]
> I noted the scenario and issue faced with that. 
> 
> 
> 
> When searching the query log, you need to search each log from each dns server (always a guess) and since I have sync enabled. DNS 1 is master. So if I make a change on dns 2, dns 1 will override dns 2. 
> 
> 
> 
> Even though let’s say the client/device is on dns 2, I now need to make the change on DNS 1 so the sync can occur over. Not a good solution and having more than 1 dns server is very common in all types or organizations 

What are you using to sync?

There's no official sync tool.

Just make the applicable changes on both AGH instances.

**Reply by iptvcld** [reply]
AdGuardHome-sync docker by Linux server.io. In a multi dns setup, you need sync and they need to be the same so clients can use any of the 2 dns servers at any time and have the rules applied.

**Reply by Dynamic5912** [reply]
> AdGuardHome-sync docker by Linux server.io. In a multi dns setup, you need sync and they need to be the same so clients can use any of the 2 dns servers at any time and have the rules applied. 

Then you need to ask the maintainer of the sync tool. 

This is not AGH's issue to fix.

**Reply by iptvcld** [reply]
> > AdGuardHome-sync docker by Linux server.io. In a multi dns setup, you need sync and they need to be the same so clients can use any of the 2 dns servers at any time and have the rules applied.
> 
> Then you need to ask the maintainer of the sync tool.
> 
> This is not AGH's issue to fix.

It’s not a sync issue I am referring to. You had asked what I am using to sync.

**Reply by Dynamic5912** [reply]
> > > AdGuardHome-sync docker by Linux server.io. In a multi dns setup, you need sync and they need to be the same so clients can use any of the 2 dns servers at any time and have the rules applied.
> 
> > 
> 
> > Then you need to ask the maintainer of the sync tool.
> 
> > 
> 
> > This is not AGH's issue to fix.
> 
> 
> 
> It’s not a sync issue I am referring to. You had asked what I am using to sync. 

So what are you asking?

**Reply by iptvcld** [reply]
It’s in my post.. everything you have asked is in my original post. Sorry theta your not understanding

**Reply by Dynamic5912** [reply]
> It’s in my post.. everything you have asked is in my original post. Sorry theta your not understanding 

OK

_"Should I be able to see all the blocks and activity from DNS server 1?"_

No you can't. Only blocks that happened on that server instance.

**Reply by iptvcld** [reply]
And this is where the issue is, in a multi dns setup, there should be an option to view logs from one place and not having to log into multiple servers to find the query log info I am after.

**Reply by Dynamic5912** [reply]
> And this is where the issue is, in a multi dns setup, there should be an option to view logs from one place and not having to log into multiple servers to find the query log info I am after. 

But AGH doesn't support multiple servers and cross-instance syncing. 

It's done by a third-party.

Again, not AGH's problem or issue to fix.

**Reply by iptvcld** [reply]
Again,  not talking about sync. Let’s not confuse the ask with this. 

But if AGH does not support multi instances then that’s different.. and really should be something that’s looked into as it’s a common setup to have more than one DNS server in an environment for redundancy and the logs for all instances should be in one location instead of guessing what server the client is being served from. 

I will leave this open in case others have any suggestions or idea to share.

**Reply by Dynamic5912** [reply]
> Again,  not talking about sync. Let’s not confuse the ask with this. 
> 
> 
> 
> But if AGH does not support multi instances then that’s different.. and really should be something that’s looked into as it’s a common setup to have more than one DNS server in an environment for redundancy and the logs for all instances should be in one location instead of guessing what server the client is being served from. 
> 
> 
> 
> I will leave this open in case others have any suggestions or idea to share. 

It's called AdGuard HOME.

Not AdGuard Business or Enterprise.

An everyday home user will have one instance setup.

You're asking for/expecting something that isn't on the radar for the everyday home user.

**Reply by iptvcld** [reply]
It’s also common for a home user to have multi dns setup. Always good to have redundant services no matter at home or enterprise. It’s just good practice.

**Comment by iptvcld** [comment]


**Reply by Dynamic5912** [reply]
That's not supported nor a feature that's in-demand or widely requested.

**Reply by iptvcld** [reply]
My company has 4 DNS servers. There is no way they are checking each of the 4 servers one at a time to review logs for a client that let’s say is using dns 3.

**Reply by Dynamic5912** [reply]
> My company has 4 DNS servers. There is no way they are checking each of the 4 servers one at a time to review logs for a client that let’s say is using dns 3. 

That's fine for your company.

This isn't AGH's issue to fix.

**Reply by iptvcld** [reply]
I was asking what others are doing for this type of setup. But it turned into a lesson on how dns services should be set up.

**Reply by Dynamic5912** [reply]
> I was asking what others are doing for this type of setup. But it turned into a lesson on how dns services should be set up. 

You're using a third-party sync tool. No one here can help with that.

Logs on AGH server 1 are kept on AGH1. 

Likewise for the other servers that are logged on their respective logs.

**Comment by he-pennypacker** [comment]
Completely agree that this is not AGH's issue to fix, however others have come up with a similar need and developed solutions such as this to keep multiple instances in sync:

https://github.com/bakito/adguardhome-sync

@iptvcld this could suit your purposes?

**Reply by iptvcld** [reply]
It’s not a matter of keeping two instances in sync.. 

When running multiple dns servers, the endpoint will randomly pick a dns server to query and if you want to allow a blocked query for an endpoint, it’s always a guessing game to first determine which dns server is the endpoint on. Then once you check both servers you will discover where to add the allow on. 

There should be a better way to viewing the logs instead of checking multiple servers when people have multi dns servers setup. 

This is not a sync issue I am bringing up..
